### PR TITLE
Laboratory work #1, Gusev Andrey - 22FPL1

### PR DESCRIPTION
**Pull request by AndreyGusev1**

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -5,6 +5,18 @@
 
 
 def tokenize(text: str) -> list[str] | None:
+    if isinstance(text, str):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:8]: Try to use early return. In this case you do not need `esle` statement at all and the function is more readable

**Comment by artyomtugaryov** [comment]
Good start, but you should accelerate...

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (AndreyGusev1) [reply lab_1_classify_by_unigrams/main.py]: Done

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -11,6 +11,15 @@ def tokenize(text: str) -> list[str] | None:
...
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    punc = "!#$%&'()*+,-./:;<=>?@[\]^_`{|}~1234567890"
+    tokens = "".join(text.split()).lower()
+    for el in tokens:
+        if el in punc:
+            tokens = tokens.replace(el, '')
+    tokens = list(tokens)
```
[the remark this answers is not in the document]
Comment (AndreyGusev1) [reply lab_1_classify_by_unigrams/main.py]: Done

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -28,6 +45,11 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
+    if not isinstance(language, str) or not isinstance(text, str):
+        return None
+    tokens = tokenize(text)
+    language_profile = {'name': language, 'freq': calculate_frequencies(tokens)}
```
[the remark this answers is not in the document]
Comment (AndreyGusev1) [reply lab_1_classify_by_unigrams/main.py]: Done

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -11,6 +11,14 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    punc = "!#$%&'()*+,-./:;<=>?@[\]^_`{|}~1234567890"
+    tokens = "".join(text.split()).lower()
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:17]: looks like you do not need to call `join`

Comment (AndreyGusev1) [reply lab_1_classify_by_unigrams/main.py]: Done

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -11,6 +11,14 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    punc = " !#$%&'\'()*+,-./:;<=>?@[\\]^_`{|}~1234567890"
+    tokens = text.lower()
+    for el in tokens:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:18]: try to use list comprehension instead of creating a new string in loop

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**